scripts/splitSources.py

```diff
@@ -59,25 +59,21 @@ def writeSourceToFile(lines):
     filePath = sys.argv[1]
     sys.excepthook = uncaught_exception_hook
 
+    # decide if file has multiple sources
     try:
-        # decide if file has multiple sources
         with open(filePath, mode='r', encoding='utf8', newline='') as f:
             lines = f.read().splitlines()
-        if len(lines) >= 1 and lines[0][:12] == "==== Source:":
-            hasMultipleSources = True
-            writeSourceToFile(lines)
-
-        if hasMultipleSources:
-            srcString = ""
-            for src in createdSources:
-                srcString += src + ' '
-            print(srcString)
-            sys.exit(0)
-        else:
-            sys.exit(1)
+    except UnicodeDecodeError as error:
+        print(f'Found UnicodeDecodeError: {error.reason}. Will now attempt to read file using "latin-1" encoding.')
+        with open(filePath, mode='r', encoding='latin-1', newline='') as f:
+            lines = f.read().splitlines()
```

Comment:
I don't think this is the right solution here. latin-1 looks like it works because because there are no invalid byte sequences in it but it will mangle the file when you then write it back as utf-8. I.e. it will write the utf-8 encoding of the latin-1 character instead of writing back the original bytes it read. As a result we'll not get the same content back when we later read the file as utf-8.

The right solution is to read the file in binary mode.

Reply:
[Guess I should read my notifications first before I start replying](https://github.com/ethereum/solidity/issues/9710#issuecomment-1342933771)